cmd/local-gadget/trace/trace.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2022 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package trace
+
+import (
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/cilium/ebpf"
+	"github.com/spf13/cobra"
+
+	commontrace "github.com/kinvolk/inspektor-gadget/cmd/common/trace"
+	commonutils "github.com/kinvolk/inspektor-gadget/cmd/common/utils"
+	"github.com/kinvolk/inspektor-gadget/cmd/local-gadget/utils"
+	containercollection "github.com/kinvolk/inspektor-gadget/pkg/container-collection"
+	"github.com/kinvolk/inspektor-gadget/pkg/gadget-collection/gadgets/trace"
+	"github.com/kinvolk/inspektor-gadget/pkg/gadgets"
+	localgadgetmanager "github.com/kinvolk/inspektor-gadget/pkg/local-gadget-manager"
+	eventtypes "github.com/kinvolk/inspektor-gadget/pkg/types"
+)
+
+// TraceGadget represents a gadget belonging to the trace category.
+type TraceGadget[Event commontrace.TraceEvent] struct {
+	commonFlags *utils.CommonFlags
+	parser      commontrace.TraceParser[Event]
+	runTracer   func(*ebpf.Map, gadgets.DataEnricher, func(Event)) (trace.Tracer, error)
+}
+
+// Run runs a TraceGadget and prints the output after parsing it using the
+// TraceParser's methods.
+func (g *TraceGadget[Event]) Run() error {
+	localGadgetManager, err := localgadgetmanager.NewManager(g.commonFlags.RuntimeConfigs)
+	if err != nil {
+		return commonutils.WrapInErrManagerInit(err)
+	}
+	defer localGadgetManager.Close()
+
+	// TODO: Improve filtering, see further details in
+	// https://github.com/kinvolk/inspektor-gadget/issues/644.
+	containerSelector := containercollection.ContainerSelector{
+		Name: g.commonFlags.Containername,
+	}
+
+	traceName, err := localGadgetManager.TracerAddTrace(containerSelector)
+	if err != nil {
+		return commonutils.WrapInErrManagerAddTracer(err)
+	}
+	defer localGadgetManager.TracerRemoveTrace(traceName)
+
+	// Get mount namespace map to filter by containers
+	mountnsmap, err := localGadgetManager.TracerMountNsMap(traceName)
+	if err != nil {
+		return commonutils.WrapInErrManagerGetMountNsMap(err)
+	}
```

Comment:
I was expecting it to capture all the events on the host if not container name was passed. I think it's fine but I also think  https://github.com/kinvolk/inspektor-gadget/issues/108 becomes more relevant because local-gadget should be able to trace events directly on the host. /cc @alban do you have any feedback?

Reply:
Yeah, as soon as we start using the trace-collector, we start ignoring the events on the host. I think we could create a flag `--host` or something similar to include them.